httpstats/context.go

```diff
@@ -0,0 +1,62 @@
+package httpstats
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/segmentio/stats"
+)
+
+// RequestWithTags returns a shallow copy of req with its context changed with this provided tags
+// so the they can be used later during the RoundTrip in the metrics recording.
+// The provided ctx must be non-nil.
+func RequestWithTags(req *http.Request, tags ...stats.Tag) *http.Request {
```

Comment:
Should we make a copy of the slice here?  Does it matter since we're not being passed a pointer?  I imagine this slice will be safe to take the address of because it was copied on the stack when calling RequestWithTags, but we might want to make a copy in case the caller changes the underlying slice.